src/tests/end-to-end/conftest.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from distutils.dir_util import copy_tree
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import pytest
+from filelock import FileLock
+from pyk.kore.rpc import kore_server
+
+from kontrol.foundry import Foundry, init_project
+from kontrol.kompile import BuildOptions, foundry_kompile
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+    from typing import Final
+
+    from pyk.kore.rpc import KoreServer
+    from pytest import TempPathFactory
+
+
+@pytest.fixture(scope='module')
+def server(foundry: Foundry, no_use_booster: bool) -> Iterator[KoreServer]:
+    llvm_definition_dir = foundry.out / 'kompiled' / 'llvm-library' if not no_use_booster else None
+    kore_rpc_command = ('kore-rpc-booster',) if not no_use_booster else ('kore-rpc',)
+
+    yield kore_server(
+        definition_dir=foundry.kevm.definition_dir,
+        llvm_definition_dir=llvm_definition_dir,
+        module_name=foundry.kevm.main_module,
+        command=kore_rpc_command,
+        smt_timeout=500,
+        smt_retry_limit=10,
+        fallback_on=None,
+        interim_simplification=None,
+        no_post_exec_simplify=None,
+    )
+
+
+TEST_DATA_DIR: Final = (Path(__file__).parent / 'test-data').resolve(strict=True)
+
+
+@pytest.fixture(scope='session')
+def foundry(foundry_root_dir: Path | None, tmp_path_factory: TempPathFactory, worker_id: str) -> Foundry:
```

Comment:
Is this copy-pasted code? If so, the duplication has to be factored out.

Reply:
@tothtamas28 The two fixtures are a bit different.

- `end-to-end` initializes a new project without dependency management and using the latest versions available, while `integration`installs in the current `foundry` project specific versions of forge-std and kontrol-cheatcodes.
-  the `foundry_kompile` options are simplified.
- project directories and the specific files copied differ,because the setups are for different testing scopes.

It should be possible to merge these two fixtures together. But I do not know on which condition to branch the behaviours.

Reply:
The fixture could also return a function that initializes the project under some (by default, temporary) directory. Would that make it easier to remove some duplication?